### PR TITLE
General exporter and builderConstructor improvements

### PR DESCRIPTION
**Pull request by Pacman99**
This is some fixups and improvments to the api for the overlaysFromChannelsExporter and packagesFromOverlaysBuilderConstructor. 

Have overlays exporter take `self.pkgs` and `inputs`. The former is so nixpkgs sets are instantiated and the overlay lists are ready. The latter is to auto filter any overlays that were taken from inputs. Its not good to export things that weren't originated in your flake and you don't plan on maintaining.
 
Change packages builderConstructor to just take an attrset of overlays and the builder can then pull out packages defined in overlays from the channels set.

### CODE REVIEW

**Comment by Pacman99** [comment]
Not sure whats up with the nixpkgs-fmt check, I ran it on my system and the file doesn't get changed.

**Comment by Pacman99** [comment]
This depends on https://github.com/numtide/flake-utils/pull/31

File: packagesFromOverlaysBuilderConstructor.nix [hunk 1]
```diff
@@ -1,89 +1,76 @@
 { flake-utils-plus }:
 let
 
-  overlayFromPackagesBuilderConstructor = channels:
+  overlayFromPackagesBuilderConstructor = overlays:
```
Comment (blaggacao) [inline packagesFromOverlaysBuilderConstructor.nix:4]: If you don't mind, would you cover up my misnomer? It should be `packagesFromOverlaysBuilderConstructor`. I probably was convincingly conaitent woth that error down the file.

Sry!

:wink:

File: packagesFromOverlaysBuilderConstructor.nix [hunk 2]
```diff
@@ -1,89 +1,76 @@
...
-          "development/vim" = { ... };
-          };
-
-          overlays' = [
-          "development"
-          ];
-
-          overlays' will pass both pkgs' through the sieve.
+            self.packages = {
+              firefox = prev.override { privacySupport = true; };
```
Comment (blaggacao) [inline packagesFromOverlaysBuilderConstructor.nix:33]: I reckon technically, it couldn't be `prev.`, I think you wanted to put in a placeholder that represents a `<derivation-with-overlay-applied>`

File: overlaysFromChannelsExporter.nix [hunk 3]
```diff
@@ -31,14 +33,23 @@ let
 
       **/
     let
-      inherit (builtins) mapAttrs attrNames concatMap listToAttrs;
+      inherit (builtins) mapAttrs foldl' filter attrNames attrValues concatMap listToAttrs elem;
       nameValuePair = name: value: { inherit name value; };
 
+      # overlays aren't system specific so arch doesn't matter here
+      channels = pkgs.x86_64-linux;
```
Comment (blaggacao) [inline overlaysFromChannelsExporter.nix:40]: Could we rather use the first element of whatever aarch is defined there? That would appear to me as a safer way which could not enter the failure mode where x86 is not among the user-supported architecture.

**Review by blaggacao — APPROVED** [review]
As an extremely promising API fix, this should make it into the upcoming release.

File: packagesFromOverlaysBuilderConstructor.nix (hunk 2 above)
Comment (Pacman99) [reply packagesFromOverlaysBuilderConstructor.nix]: good point, just wrote out what I was going for in text

File: packagesFromOverlaysBuilderConstructor.nix (hunk 1 above)
Comment (Pacman99) [reply packagesFromOverlaysBuilderConstructor.nix]: fixed to follow the same naming

File: overlaysFromChannelsExporter.nix (hunk 3 above)
Comment (Pacman99) [reply overlaysFromChannelsExporter.nix]: I used head and attrValues to fix this and just get the first system in alphabetical order. It doesn't really matter what system it uses, just needs to get the instantiated.

**Comment by Pacman99** [comment]
This should be ready to go, I just changed the commit to update flake-utils, since the fixup PR has been merged.

File: overlaysFromChannelsExporter.nix [hunk 4]
```diff
@@ -31,14 +33,24 @@ let
...
+      # overlays can be safely evaluated on any arch
+      channels = head (attrValues pkgs);
+
       pathStr = path: builtins.concatStringsSep "/" path;
 
       channelNames = attrNames channels;
       overlayNames = overlay: attrNames (overlay null null);
 
+      /* workaround to detect overlays owned by this flake, by filtering out overlays from others */
+      # get all overlays from inputs
```
Comment (gytis-ivaskevicius) [inline overlaysFromChannelsExporter.nix:49]: Whats up with different type of comments :D

File: packagesFromOverlaysBuilderConstructor.nix [hunk 5]
```diff
@@ -1,92 +1,81 @@
...
-          (_: pkgs:
-            filterAttrs
-              (pkgName:
-                any (overlayName: hasPrefix pkgName overlayName) flattendOverlaysNames
-              )
-              pkgs
-          )
-          flattenedPackages;
+        # flatten any nested sets with with "/" delimiter
+          # filter out packages that don't meet the flake check conditions
```
Comment (gytis-ivaskevicius) [inline packagesFromOverlaysBuilderConstructor.nix:74]: Why doesn't nixpkgs-fmt pick up these comments alignment :/

**Review by gytis-ivaskevicius — APPROVED** [review]
Looks great, tho Ill need to try it out with odd values to see if it is actually working as expected.

Feel free to merge it whenever you think its ready

File: overlaysFromChannelsExporter.nix (hunk 4 above)
Comment (Pacman99) [reply overlaysFromChannelsExporter.nix:49]: I was going for a header, but I think I'm going to move that comment to the top next to the `inputs` argument description. So its clear why its needed.

File: packagesFromOverlaysBuilderConstructor.nix (hunk 5 above)
Comment (Pacman99) [reply packagesFromOverlaysBuilderConstructor.nix]: actually `nixpkgs-fmt` forces that formatting, I tried to put them both on the same indent, but fmt kept staggering them. I condensed it to just one comment.

**Pull request merged**